### PR TITLE
remove column from logs if only unchanged elements exists

### PR DESCRIPTION
**Pull request by krzysztofruszczynski**
- remove column from logs if only unchanged elements exists

### CODE REVIEW

File: src/DataHandling/Logs/DataDogAudit.php [hunk 1]
```diff
@@ -127,6 +127,27 @@ protected function getCurrentVersionElement($currentVersion, $diffElement)
...
+     * @return array subsequent elements are diff for each version, filtered
+     */
+    protected function removeColumnIfOnlyUnchanged($diffArray)
+    {
+        foreach ($diffArray as $versionKey => $versionValue) {
+            foreach ($versionValue as $columnName => $columnValues) {
+                if (is_array($diffArray[$versionKey][$columnName])
+                        && isset($diffArray[$versionKey][$columnName][self::KEY_UNCHANGED])
+                        && !isset($diffArray[$versionKey][$columnName][self::KEY_ADD])
+                        && !isset($diffArray[$versionKey][$columnName][self::KEY_REMOVE])) {
```
Comment (SimonHeimberg) [inline src/DataHandling/Logs/DataDogAudit.php]: question: is `isset()` sufficent? So is the element never an empty array (`array()`)? Or should `!empty()` be used instead?

Comment (krzysztofruszczynski) [reply src/DataHandling/Logs/DataDogAudit.php]: Logic in getCurrentVersionElement($currentVersion, $diffElement) (especially line 117) ensures that that there is no empty array with added or removed elements (added array is unset when empty, removed array is created only if there are real elements removed).

File: src/DataHandling/Logs/DataDogAudit.php [hunk 2]
```diff
@@ -87,13 +87,28 @@ protected function getAssociations($entity)
     /**
      * Method creating version key (to group table entries associated with the same user action).
      * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion entity for which version key is created
+     * @param array $diffArray subsequent elements are diff for each version
```
Comment (SimonHeimberg) [inline src/DataHandling/Logs/DataDogAudit.php]: stile hint: empty comment line (`\n*\n`) before `@param` and `@return`.
Not for fixing, but for doing next time.

File: src/DataHandling/Logs/DataDogAudit.php [hunk 3]
```diff
@@ -87,13 +87,28 @@ protected function getAssociations($entity)
...
      * @return string version key (timestamp and user id)
      */
-    protected function getVersionKey($currentVersion)
+    protected function getVersionKey($currentVersion, $diffArray)
     {
         $versionTimestamp = $currentVersion->getLoggedAt()->getTimestamp();
+        $versionTimestampBefore = $currentVersion->getLoggedAt()->getTimestamp() - 1; // 1 second before by the same user
+        $versionTimestampAfter = $currentVersion->getLoggedAt()->getTimestamp() + 1; // 1 second after by the same user
```
Comment (SimonHeimberg) [inline src/DataHandling/Logs/DataDogAudit.php]:
```php
$versionTimestampBefore = $versionTimestamp - 1; // 1 second before
$versionTimestampAfter = ...
```

is clearer

File: src/DataHandling/Logs/DataDogAudit.php (hunk 1 above)
Comment (SimonHeimberg) [reply src/DataHandling/Logs/DataDogAudit.php]: 👍 fine in this case

**Pull request merged**